eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -274,28 +288,45 @@ object Graph {
-    RichWeight(totalCost, prev.length + 1, totalCltv, totalWeight)
+  private def addEdgeWeight(sender: PublicKey, edge: GraphEdge, prev: RichWeight, currentBlockHeight: Long, weightRatios: Either[WeightRatios, HeuristicsConstants], includeLocalChannelCost: Boolean): RichWeight = {
+    val totalAmount = if (edge.desc.a == sender && !includeLocalChannelCost) prev.amount else addEdgeFees(edge, prev.amount)
+    val fee = totalAmount - prev.amount
+    val totalFees = prev.fees + fee
+    val cltv = if (edge.desc.a == sender && !includeLocalChannelCost) CltvExpiryDelta(0) else edge.update.cltvExpiryDelta
+    val totalCltv = prev.cltv + cltv
+    weightRatios match {
+      case Left(weightRatios) =>
+        val hopCost = nodeFee(weightRatios.hopCost.feeBase, weightRatios.hopCost.feeProportionalMillionths, prev.amount)
+        import RoutingHeuristics._
+
+        // Every edge is weighted by funding block height where older blocks add less weight. The window considered is 1 year.
+        val channelBlockHeight = ShortChannelId.coordinates(edge.desc.shortChannelId).blockHeight
+        val ageFactor = normalize(channelBlockHeight, min = (currentBlockHeight - BLOCK_TIME_ONE_YEAR).toDouble, max = currentBlockHeight.toDouble)
+
+        // Every edge is weighted by channel capacity, larger channels add less weight
+        val edgeMaxCapacity = edge.capacity.toMilliSatoshi
+        val capFactor = 1 - normalize(edgeMaxCapacity.toLong.toDouble, CAPACITY_CHANNEL_LOW.toLong.toDouble, CAPACITY_CHANNEL_HIGH.toLong.toDouble)
+
+        // Every edge is weighted by its cltv-delta value, normalized
+        val cltvFactor = normalize(edge.update.cltvExpiryDelta.toInt, CLTV_LOW, CLTV_HIGH)
+
+        // NB we're guaranteed to have weightRatios and factors > 0
+        val factor = weightRatios.baseFactor + (cltvFactor * weightRatios.cltvDeltaFactor) + (ageFactor * weightRatios.ageFactor) + (capFactor * weightRatios.capacityFactor)
+        val totalWeight = prev.weight + (fee + hopCost).toLong * factor
+        RichWeight(totalAmount, prev.length + 1, totalCltv, 1.0, totalFees, totalWeight)
+      case Right(heuristicsConstants) =>
+        val hopCost = nodeFee(heuristicsConstants.hopCost.feeBase, heuristicsConstants.hopCost.feeProportionalMillionths, prev.amount)
+        val cltv = if (edge.desc.a == sender && !includeLocalChannelCost) CltvExpiryDelta(0) else edge.update.cltvExpiryDelta
+        val totalCltv = prev.cltv + cltv
+        val riskCost = totalAmount.toLong * cltv.toInt * heuristicsConstants.lockedFundsRisk
+        // If the edge was added by the invoice, it is assumed that it can route the payment.
+        // If we know the balance of the channel, then we will check separately that it can relay the payment.
+        val successProbability = if (edge.update.chainHash == ByteVector32.Zeroes || edge.balance_opt.nonEmpty) 1.0 else 1.0 - totalAmount.toLong.toDouble / (1000 * edge.capacity.toLong.toDouble)
```

Comment:
You should use `edge.capacity.toMilliSatoshi` instead of multiplying by 1000, it would make it easier to see that we're comparing apples to apples.

Reply:
Side note: it would actually be interesting to do A/B testing on the probability distribution we use for the channel balance in the future, if we think this new heuristic is likely to replace the old one